Secret-agency-web/src/main/java/cz/fi/muni/pa165/secretagency/controllers/AgentsController.java

```diff
@@ -0,0 +1,107 @@
+package cz.fi.muni.pa165.secretagency.controllers;
+
+import cz.fi.muni.pa165.secretagency.ApiUris;
+import cz.fi.muni.pa165.secretagency.dto.AgentCreateDTO;
+import cz.fi.muni.pa165.secretagency.dto.AgentDTO;
+import cz.fi.muni.pa165.secretagency.enums.AgentRankEnum;
+import cz.fi.muni.pa165.secretagency.enums.LanguageEnum;
+import cz.fi.muni.pa165.secretagency.exceptions.ResourceNotFoundException;
+import cz.fi.muni.pa165.secretagency.facade.AgentFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.http.MediaType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * @author Adam Skurla (487588)
+ * Agents Rest Controller
+ */
+@RestController
+@RequestMapping(ApiUris.ROOT_URI_REST + ApiUris.ROOT_URI_AGENTS)
+public class AgentsController {
+    final static Logger logger = LoggerFactory.getLogger(AgentsController.class);
+
+    @Autowired
+    private AgentFacade agentFacade;
+
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public final List<AgentDTO> getAgents() {
+        logger.debug("rest getAgents()");
+        return this.agentFacade.getAllAgents();
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public final AgentDTO getAgent(@PathVariable("id") Long id) throws ResourceNotFoundException {
+        logger.debug("rest getAgent({})", id);
+        try {
+            return agentFacade.getAgentById(id);
+        } catch (Exception ex) {
+            throw new ResourceNotFoundException();
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public final AgentDTO createAgent(@RequestBody AgentCreateDTO agentCreateDTO) {
+        logger.debug("rest createAgent({})", agentCreateDTO);
+        Long id = this.agentFacade.createAgent(agentCreateDTO);
+        return this.agentFacade.getAgentById(id);
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
+    public final void deleteAgent(@PathVariable("id") Long id) {
+        logger.debug("rest deleteAgent({})", id);
+        this.agentFacade.deleteAgent(id);
```

Comment:
tu by to chelo nejak ohandlovat, jestli agent nema nejake mise/reporty, jinak to podle me spadne